gadgets/src/integers/int/arithmetic/div.rs

```diff
@@ -63,191 +61,80 @@ macro_rules! div_int_impl {
                 // else
                 //    !Q                      -- negative result
 
-                if other.eq(&Self::constant(0 as <$gadget as Integer>::IntegerType)) {
-                    return Err(SignedIntegerError::DivisionByZero);
+                // If `self` and `other` are both constants, return the constant results instead of generating constraints
```

Comment:
This part of the gadget has been completely rewritten following the comments and information originally provided in the document. The new gadget works in the setup mode and avoids incorrect constant allocations.

Reply:
Explain: In addition, the new code fixes a previous error. Previously, dividing -128 by 2 will result in -63 instead of -64.